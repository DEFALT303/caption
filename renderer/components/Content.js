import Drop from "../components/drop";
import ListEmpty from "../components/listEmpty";
import ListFiles from "../components/listFiles";

export default ({ searchQuery, files, onDrop, setRef }) =>
  <section>
    {searchQuery !== "" && <ListEmpty />}
    {searchQuery === "" &&
      files.length === 0 &&
      <Drop onDrop={onDrop} setRef={setRef} />}
    {files.length > 0 && <ListFiles files={files} />}

    <style jsx>{`
      section {
        height: calc(100vh - 120px);
        overflow: hidden;
        overflow-y: scroll;
      }
    `}</style>
  </section>;
