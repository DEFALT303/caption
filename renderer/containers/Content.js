import { connect } from "react-redux";

import Content from "./../components/Content";
import { dropFiles } from "./../actions";

// Utils
import { processFiles } from "../utils";

const mapStateToProps = ({ search }) => ({
  searchQuery: search.searchQuery,
  files: search.files,
  results: search.results,
  loading: search.loading,
});
const mapDispatchToProps = {
  onDrop: rawFiles => async dispatch => {
    const files = await processFiles(rawFiles);
    dispatch(dropFiles(files));
  },
};

export default connect(mapStateToProps, mapDispatchToProps)(Content);
