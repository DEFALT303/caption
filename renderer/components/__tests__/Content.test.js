import ReactTestRenderer from "react-test-renderer";
import Content from "./../Content";

jest.mock("react-dropzone", () => 'FakeDropzone');

describe("<Content />", () => {
  it("should show <ListEmpty /> when there is a query but no results", () => {
    const tree = ReactTestRenderer.create(
      <Content searchQuery="test" results={[]} />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should show <Drop /> when there is no searchQuery and no files dropped", () => {
    const tree = ReactTestRenderer.create(
      <Content searchQuery="" files={[]} />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
