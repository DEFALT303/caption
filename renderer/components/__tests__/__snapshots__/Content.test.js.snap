// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render <List /> when files dropped are dropped 1`] = `
<section
  className="jsx-1759955757 "
>
  <ul
    className="jsx-565981134"
  >
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
  </ul>
  <ul
    className="jsx-631033484"
  >
    <li
      className="jsx-372683689 "
      onClick={[Function]}
      onContextMenu={[Function]}
      onDoubleClick={[Function]}
    />
    <li
      className="jsx-372683689 "
      onClick={[Function]}
      onContextMenu={[Function]}
      onDoubleClick={[Function]}
    />
  </ul>
</section>
`;

exports[`<Content /> should render <List /> when there are results 1`] = `
<section
  className="jsx-1759955757 "
>
  <ul
    className="jsx-631033484"
  >
    <li
      className="jsx-372683689 "
      onClick={[Function]}
      onContextMenu={[Function]}
      onDoubleClick={[Function]}
    />
    <li
      className="jsx-372683689 "
      onClick={[Function]}
      onContextMenu={[Function]}
      onDoubleClick={[Function]}
    />
  </ul>
</section>
`;

exports[`<Content /> should show <Drop /> when there is no searchQuery and no files dropped 1`] = `
<section
  className="jsx-1759955757 "
>
  <div
    className="jsx-246188302"
  >
    <FakeDropzone
      activeStyle={
        Object {
          "border": "1px dashed #0095ff",
        }
      }
      onDrop={undefined}
      preventDropOnDocument={true}
      style={
        Object {
          "border": "1px dashed rgba(202, 203, 204, 1)",
          "borderRadius": "5px",
          "height": "100%",
          "transition": "all .2s ease-out",
          "width": "100%",
        }
      }
    >
      <p
        className="jsx-246188302"
      >
        Drop an episode or season...
      </p>
    </FakeDropzone>
  </div>
</section>
`;

exports[`<Content /> should show <ListEmpty /> when there is a query but no results 1`] = `
<section
  className="jsx-1759955757 "
>
  <ul
    className="jsx-565981134"
  >
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
    <li
      className="jsx-565981134"
    />
  </ul>
</section>
`;
